### PR TITLE
Automatic sync of custom channels

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

This PR add a new configuration property to unify the management of custom and vendor channels. In particular when the property is set to true:

1) custom channel are synced together with vendor channels
2) when a custom channel is added using `spacewalk-common-channels` it is automatically synced
3) the UI to set up scheduled syncronization is disabled

## GUI diff

After - Only when the configuration parameter is set to true:

Screenshot from 2022-07-12 14-21-25

- [X] **DONE**

## Documentation

- Documentation PR WIP

- [ ] **DONE**

## Test coverage
- No tests: already covered

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/18138

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
Should this unified handling not the default in "master" ?

File: java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java [hunk 1]
```diff
@@ -1308,17 +1308,27 @@ public static void addErrataToChannel(Set<Long> eids, Long cid) {
...
+        @SuppressWarnings("unchecked")
+        List<Channel> result = singleton.listObjectsByNamedQuery("Channel.findVendorChannels", Collections.emptyMap());
         if (result != null) {
             return result;
         }
         return new ArrayList<>();
     }
 
+    /**
+     * List all vendor channels (org is null)
```
[the remark this answers is not in the document]
Comment (mackdk) [reply java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java]: Fixed

File: java/code/src/com/redhat/rhn/taskomatic/task/MgrSyncRefresh.java [hunk 2]
```diff
@@ -112,9 +112,15 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
...
+                    final TaskomaticApi taskoApi = new TaskomaticApi();
+
                     log.debug("Scheduling synchronization of all vendor channels");
-                    new TaskomaticApi().scheduleSingleRepoSync(
-                            ChannelFactory.listVendorChannels());
+                    taskoApi.scheduleSingleRepoSync(ChannelFactory.listVendorChannels());
+
+                    if (ConfigDefaults.get().isCustomChannelManagementUnificationEnabled()) {
+                        log.debug("Scheduling synchronization of all custom channels");
+                        taskoApi.scheduleSingleRepoSync(ChannelFactory.listCustomChannels());
```
[the remark this answers is not in the document]
Comment (mackdk) [reply java/code/src/com/redhat/rhn/taskomatic/task/MgrSyncRefresh.java]: I changed the query to retrieve only the custom channels that have at least one repository configured, so now we should not schedule unneeded repo sync runs.

**Comment by mackdk** [comment]
> Should this unified handling not the default in "master" ?

I changed the code to have true as default value for the new configuration property. When merging to the other branches, I will revert the default to false, so that we do not suddenly change behaviour for existing installation.

File: java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java [hunk 3]
```diff
@@ -355,6 +355,11 @@ public class ConfigDefaults {
      */
     public static final String DEFAULT_DOCS_LOCALE = "web.docs_locale";
 
+    /**
+     * Specify if custom channels are synced together with vendor channels
+     */
+    public static final String UNIFY_CUSTOM_CHANNEL_MANAGEMENT = "java.unify_custom_channel_management";
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/conf/ConfigDefaults.java:361]: You can also specify the default in the config-defaults file: java/conf/rhn_java.conf

Add the key there with true or false and add a description what this is doing.

**Comment by mcalmer** [comment]
I think the failed translation check should be just a warning that this PR modify translation, right?
Maybe you can ask Karl if this is wanted this way. We now need to ignore this "error".

**Review by mcalmer — APPROVED** [review]

**Pull request merged**